python-sdk/src/astro/sql/operators/raw_sql.py

```diff
@@ -28,11 +30,22 @@ def execute(self, context: dict) -> Any:
         result = self.database_impl.run_sql(
             sql_statement=self.sql, parameters=self.parameters
         )
+        if self.response_size == -1 and not settings.IS_CUSTOM_XCOM_BACKEND:
+            logging.warning(
+                "Using `run_raw_sql` without `response_size` can result in excessive amount of data being recorded "
+                "to the Airflow metadata database, leading to issues to the orchestration of tasks. It is possible to "
+                "avoid this problem by either setting `response_size` to a small integer or by using a custom XCom "
+                "backend."
+            )
+
         if self.handler:
             response = self.handler(result)
             if self.response_limit and len(response) > self.response_limit:
                 raise IllegalLoadToDatabaseException()  # pragma: no cover
-            return response
+            if self.response_size >= 0:
```

Comment:
..then changing this to:
```suggestion
            if self.response_size > 0:
```

Reply:
(discussion in the previous thread)